java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/SqlClusterExecutor.java

```diff
@@ -61,14 +61,25 @@
     public SqlClusterExecutor(SdkOption option, String libraryPath) throws SqlException {
         initJavaSdkLibrary(libraryPath);
 
-        SQLRouterOptions sqlOpt = new SQLRouterOptions();
-        sqlOpt.setSession_timeout(option.getSessionTimeout());
-        sqlOpt.setZk_cluster(option.getZkCluster());
-        sqlOpt.setZk_path(option.getZkPath());
-        sqlOpt.setEnable_debug(option.getEnableDebug());
-        sqlOpt.setRequest_timeout(option.getRequestTimeout());
-        this.sqlRouter = sql_router_sdk.NewClusterSQLRouter(sqlOpt);
-        sqlOpt.delete();
+        if (option.isClusterMode()) {
+            SQLRouterOptions sqlOpt = new SQLRouterOptions();
+            sqlOpt.setSession_timeout(option.getSessionTimeout());
+            sqlOpt.setZk_cluster(option.getZkCluster());
+            sqlOpt.setZk_path(option.getZkPath());
+            sqlOpt.setEnable_debug(option.getEnableDebug());
+            sqlOpt.setRequest_timeout(option.getRequestTimeout());
+            this.sqlRouter = sql_router_sdk.NewClusterSQLRouter(sqlOpt);
+            sqlOpt.delete();
```

Comment:
don't the GC handle it automatically ? @vagetablechicken

Reply:
If `swigCMemOwn==true`, it'll gc. But we've found CMem cost too much before, so in java sdk, we manually delete CMem in every place.
Just leave it, until we can remove all manul deletion.